crates/sui-move-build/src/linters/collection_equality.rs

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+//! This analysis flags situations when instances of a sui::table::Table or sui::table_vec::TableVec
+//! or sui::bag::Bag are being compared for (in)equality at this type of comparison is not very
+//! useful and DOES NOT take into consideration structural (in)equality.
+
+use move_compiler::{
+    diag,
+    diagnostics::codes::{custom, DiagnosticInfo, Severity},
+    naming::ast as N,
+    parser::ast as P,
+    shared::{CompilationEnv, Identifier},
+    typing::{ast as T, core::ProgramInfo, visitor::TypingVisitor},
+};
+
+use super::{
+    base_type, LinterDiagCategory, BAG_MOD_NAME, BAG_STRUCT_NAME, LINTER_DEFAULT_DIAG_CODE,
+    LINT_WARNING_PREFIX, SUI_PKG_NAME, TABLE_MOD_NAME, TABLE_STRUCT_NAME, TABLE_VEC_MOD_NAME,
+    TABLE_VEC_STRUCT_NAME,
+};
+
+const COLLECTIONS_EQUALITY_DIAG: DiagnosticInfo = custom(
+    LINT_WARNING_PREFIX,
+    Severity::Warning,
+    LinterDiagCategory::CollectionEquality as u8,
+    LINTER_DEFAULT_DIAG_CODE,
+    "possibly useless collections compare",
+);
+
+const COLLECTION_TYPES: &[(&str, &str, &str)] = &[
+    (SUI_PKG_NAME, BAG_MOD_NAME, BAG_STRUCT_NAME),
+    (SUI_PKG_NAME, TABLE_MOD_NAME, TABLE_STRUCT_NAME),
+    (SUI_PKG_NAME, TABLE_VEC_MOD_NAME, TABLE_VEC_STRUCT_NAME),
```

Comment:
We need `LinkedTable`, `ObjectBag`, and `ObjectTable` as well. 
And sadly `VecMap` and `VecSet`. 

I don't think we need `PriorityQueue` but we should double check